Only alias sed to /usr/lib/portage/bin/ebuild-helpers/sed if the
latter exists (which should be the case on prefix portages, but not on
normal ones). Fixes fallout from bug #370935 on non-prefix portages.

--- python-updater
+++ python-updater
@@ -488,8 +488,15 @@
 	shift
 done
 
+# On a prefix portage we need to use sed from ebuild-helpers/ instead
+# of $PATH. On non-prefix portages ebuild-helpers/sed does not exist.
+# Prepending ebuild-helpers/ to $PATH uses too many tools from there.
+# So alias sed to ebuild-helpers/sed if it exists:
 if [[ "${PMS_COMMAND[${PMS_INDEX}]}" == "emerge" ]]; then
-	alias sed="$(/usr/bin/python -c "import os; print(os.path.dirname(os.path.realpath('$(type -p emerge)')))")/ebuild-helpers/sed"
+	sed="$(/usr/bin/python -c "import os; print(os.path.dirname(os.path.realpath('$(type -p emerge)')))")/ebuild-helpers/sed"
+	if [[ -x "${sed}" ]]; then
+		alias sed="${sed}"
+	fi
 fi
 
 if has_version "=dev-lang/python-2*"; then
